src/codegen/riscv64/assembler-riscv64.cc

```diff
@@ -304,6 +304,17 @@ bool Assembler::IsJalr(Instr instr) {
   return (instr & kBaseOpcodeMask) == JALR;
 }
 
+bool Assembler::IsLui(Instr instr) { return (instr & kBaseOpcodeMask) == LUI; }
+bool Assembler::IsAddiw(Instr instr) {
+  return (instr & (kBaseOpcodeMask | kFunct3Mask)) == RO_ADDIW;
+}
+bool Assembler::IsAddi(Instr instr) {
+  return (instr & (kBaseOpcodeMask | kFunct3Mask)) == RO_ADDI;
+}
+bool Assembler::IsSlli(Instr instr) {
+  return (instr & (kBaseOpcodeMask | kFunct3Mask)) == RO_SLLI;
+}
```

Comment:
These are a good addition. Let's update `target_address_at` and `set_target_value_at` to use these.